### PR TITLE
Add sentry

### LINKED ISSUE #458: Add Sentry support for all non-prod builds
We should add Sentry support to the frontend, but only enable it on non-prod builds.

All engineering contributors of Fractal have been added to the new Sentry team, so anyone should be able to pick this issue up.

https://sentry.io/organizations/decent-1k/projects/fractal-interface/?project=6688050

### PR DESCRIPTION
**Pull request by Da-Colon**
## Description

Adds Sentry and Sentry support to Fractal app. This will help collect data on errors that occurs on the dev site. There will need to be further work to capture other errors (see Next Steps)

## Notes

- This PR only covers the installation and setup of Sentry and Adds a ErrorBoundary Screen which will catch app 

## Next Steps:
- Adding sentry to any places that are catching errors.
- https://docs.sentry.io/platforms/javascript/guides/react/usage/
## Issue (if applicable)
closes #458 

## Testing

Error Boundary:
The Error Boundary is a little hard to test on a preview, but this will show up when there is a rendering issues causing the app to crash.

Sentry:
I have tested this myself and again will be hard to force a problem on the preview. but I have setup the IPFS and the develop websites to connect and report to sentry
## Screenshots (if applicable)
image

### CODE REVIEW

File: src/index.tsx [hunk 1]
```diff
@@ -3,39 +3,56 @@ import { createRoot } from 'react-dom/client';
...
+Sentry.init({
+  dsn: process.env.REACT_APP_SENTRY_DSN || '',
+  integrations: [new BrowserTracing()],
+
+  // Set tracesSampleRate to 1.0 to capture 100%
+  // of transactions for performance monitoring.
+  // We recommend adjusting this value in production
+  tracesSampleRate: 1.0,
+});
+
```
Comment (adamgall) [inline src/index.tsx:31]: When this is initialized with an empty string `dsn`, what happens?

File: .github/workflows/release-ipfs-dev.yaml [hunk 2]
```diff
@@ -48,6 +48,7 @@ jobs:
           REACT_APP_GNOSISWRAPPER_ADDRESSES: ${{ secrets.REACT_APP_GNOSISWRAPPER_ADDRESSES }}
           REACT_APP_GNOSISSAFEFACTORY_ADDRESSES: ${{ secrets.REACT_APP_GNOSISSAFEFACTORY_ADDRESSES }}
           REACT_APP_GNOSISSAFE_ADDRESSES: ${{ secrets.REACT_APP_GNOSISSAFE_ADDRESSES }}
+          REACT_APP_REACT_APP_SENTRY_DSN: ${{ secrets.REACT_APP_GNOSISSAFE_ADDRESSES }}
```
Comment (adamgall) [inline .github/workflows/release-ipfs-dev.yaml:51]: Can you please add this ENV to the committed `.env` file

**Review by adamgall — CHANGES_REQUESTED** [review]

**Comment by adamgall** [comment]
Regarding your comment about next steps, (and reading their docs), it seems like Sentry won't automatically report errors unless they're uncaught?

What does that mean we should do...

File: src/components/ErrorFallback.tsx [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+    <div className="flex flex-col bg-fixed bg-image-pattern bg-cover h-screen justify-center">
+      <div className="container">
+        <ContentBox>
+          <div className="flex gap-2 items-center">
+            <Alert />
+            <div className="text-gray-25">Oops!</div>
+          </div>
+          <div className="text-sm text-gray-50 text-mono">There was an unexpected error</div>
+          <InputBox>
+            <div className="text-md text-gray-25 text-mono">{error.message}</div>
```
[the remark this answers is not in the document]
Comment (adamgall) [reply src/components/ErrorFallback.tsx:24]: I think I agree with @mudrila here, as long as we're sure the actual error that this `ErrorBoundary` `fallback` component is catching is being sent to Sentry.

File: src/index.tsx (hunk 1 above)
Comment (Da-Colon) [reply src/index.tsx:31]: App loads normally with no errors or anything. As far as I've read everything else should continue working normally locally

File: src/components/ErrorFallback.tsx [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+          </InputBox>
+          <div>
+            <PrimaryButton
+              label="Reload"
+              className="my-2"
+              onClick={() => window.location.reload()}
```
[the remark this answers is not in the document]
Comment (Da-Colon) [reply src/components/ErrorFallback.tsx:30]: Just tried it, I get this error. 

image

File: src/components/ErrorFallback.tsx (hunk 3 above)
[the remark this answers is not in the document]
Comment (Da-Colon) [reply src/components/ErrorFallback.tsx:24]: image
Definitely working. Yeah I'll remove the error message. While the User should never really see this page. We may need to add something a little more to this error screen.

**Review by adamgall — APPROVED** [review]
I dig it, thanks @Da-Colon

File: src/components/ErrorFallback.tsx (hunk 3 above)
[the remark this answers is not in the document]
Comment (adamgall) [reply src/components/ErrorFallback.tsx]: It looks good for now, thanks!

**Comment by adamgall** [comment]
@Da-Colon

One more thing to at least touch upon here, is: what is expected for capturing Sentry errors in places where we're _already_ catching errors and `console.error`ing them / showing them as a toast?

**Comment by Da-Colon** [comment]
In the usage does (link in the desciprtion). we would simply add to our changes. 
```ts
import * as Sentry from "@sentry/react";

try {
  aFunctionThatMightFail();
} catch (err) {
  Sentry.captureException(err);
}
```

**Comment by adamgall** [comment]
> In the usage docs (link in the desciprtion). we would simply add to our catches.
> 
> ```ts
> import * as Sentry from "@sentry/react";
> 
> try {
>   aFunctionThatMightFail();
> } catch (err) {
>   Sentry.captureException(err);
> }
> ```

@Da-Colon 

This seems like a lot of work to do, probably another PR where all of those exceptions in the existing app which are currently being caught and logged, are caught/logged/sent to Sentry. Maybe there's an "error handling" function added to the app which does all of those side-effects.

**Comment by Da-Colon** [comment]
> This seems like a lot of work to do, probably another PR where all of those exceptions in the existing app which are currently being caught and logged, are caught/logged/sent to Sentry. Maybe there's an "error handling" function added to the app which does all of those side-effects.

Yessir I agree, I'll add getting this card created to my todo list

**Pull request merged**